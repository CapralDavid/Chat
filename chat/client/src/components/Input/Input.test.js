import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Input from './Input.js';


//проверить как ведет себя элемент при onChange, onKeyPress
//где то был пример как чекает когда наводят мышкой

describe('test render Input empty', () => {

    test('render correct: Input Empty', () => {

          const component = renderer.create(
              <Router>
              <Input></Input>
              </Router>
            );
        
          let treeMessages = component.toJSON();
          expect(treeMessages).toMatchSnapshot();
        
        }); 

        test('render correct: Input wirh msg', () => {
            const msgValue = "hello guys"
            const nameOne = ""
              const component = renderer.create(
                  <Router>
                  <Input value = {msgValue}></Input>
                  </Router>
                );
            
              let treeMessages = component.toJSON();
              expect(treeMessages).toMatchSnapshot();
            
            }); 





  
  
  
  
  });