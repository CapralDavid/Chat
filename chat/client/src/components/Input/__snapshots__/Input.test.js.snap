// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render Input empty render correct: Input Empty 1`] = `
<form
  className="form"
>
  <input
    className="input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="your message..."
    type="text"
  />
  <button
    className="sendButton"
    onClick={[Function]}
  >
    Send
  </button>
</form>
`;

exports[`test render Input empty render correct: Input wirh msg 1`] = `
<form
  className="form"
>
  <input
    className="input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="your message..."
    type="text"
  />
  <button
    className="sendButton"
    onClick={[Function]}
  >
    Send
  </button>
</form>
`;
